zipline/assets/_assets.pyx

```diff
@@ -187,37 +201,58 @@ cdef class Asset:
         """
         return cls(**dict_)
 
-    def _is_alive(self, dt, bool normalized):
+    def _is_alive_for_session(self, session_label):
```

Comment:
I prefer keeping this method simple - it's responsible for taking a session and returning whether the asset was alive for that session.